### PR TITLE
Improve error codes reference

### PR DESCRIPTION
**Pull request by Simran-B**
### Description

- Use error number for headline anchors instead of full headline text like in Jekyll toolchain, e.g. `#11` instead of `#11---forbidden`.

- Add a description about status/error codes, numbers, messages, and descriptions

- Fix gap in headline levels between h2 and h4

- Wrap very long headlines mid-word if they would overflow, e.g. `4011 - ERROR_UNSUPPORTED_CHANGE_IN_SMART_TO_SATELLITE_DISJOINT_EDGE_DIRECTION`

  This should only affect headlines that contain unusually long words without spaces (like error names) but not wrap mid-word for normal headlines that span multiple lines - @nikhil-varma please review this CSS change

#### Upstream PRs

- 3.10:
- 3.11:
- 3.12:

### CODE REVIEW

File: site/content/3.10/develop/error-codes-and-meanings.md [hunk 1]
```diff
@@ -2,9 +2,73 @@
...
 archetype: default
 ---
-{{% error-codes %}}
\ No newline at end of file
+{{< description >}}
+
+## Numbers, names, and descriptions of errors
+
+When an error occurs in an operation of an ArangoDB server, the
+[HTTP REST API](http/_index.md) responds to a request with with an
```
Comment (nikhil-varma) [inline site/content/3.10/develop/error-codes-and-meanings.md:17]: Double 'with'

```suggestion
[HTTP REST API](http/_index.md) responds to a request with an
```

File: site/content/3.11/develop/error-codes-and-meanings.md [hunk 2]
```diff
@@ -2,9 +2,73 @@
...
 archetype: default
 ---
-{{% error-codes %}}
\ No newline at end of file
+{{< description >}}
+
+## Numbers, names, and descriptions of errors
+
+When an error occurs in an operation of an ArangoDB server, the
+[HTTP REST API](http/_index.md) responds to a request with with an
```
Comment (nikhil-varma) [inline site/content/3.11/develop/error-codes-and-meanings.md:17]: Same here.. double 'with'

```suggestion
[HTTP REST API](http/_index.md) responds to a request with an
```

File: site/content/3.11/develop/error-codes-and-meanings.md [hunk 3]
```diff
@@ -2,9 +2,73 @@
...
+status codes defined by the HTTP protocol.
+
+A more detailed **error number** is provided in the `errorNum` attribute.
+Error numbers are specific to ArangoDB. Each ArangoDB error kind has a number,
+a unique name, an error message, and a description.
+
+The **error message** is a brief description of the error and provided in the
+`errorMessage` attribute.
+
+The **error name** is not visible in the HTTP API response. The name for the
```
Comment (nikhil-varma) [inline site/content/3.11/develop/error-codes-and-meanings.md:40]: Same structural change here as well

File: site/content/3.12/develop/error-codes-and-meanings.md [hunk 4]
```diff
@@ -2,9 +2,73 @@
...
 archetype: default
 ---
-{{% error-codes %}}
\ No newline at end of file
+{{< description >}}
+
+## Numbers, names, and descriptions of errors
+
+When an error occurs in an operation of an ArangoDB server, the
+[HTTP REST API](http/_index.md) responds to a request with with an
```
Comment (nikhil-varma) [inline site/content/3.12/develop/error-codes-and-meanings.md:17]:
```suggestion
[HTTP REST API](http/_index.md) responds to a request with an
```

File: site/content/3.12/develop/error-codes-and-meanings.md [hunk 5]
```diff
@@ -2,9 +2,73 @@
...
+status codes defined by the HTTP protocol.
+
+A more detailed **error number** is provided in the `errorNum` attribute.
+Error numbers are specific to ArangoDB. Each ArangoDB error kind has a number,
+a unique name, an error message, and a description.
+
+The **error message** is a brief description of the error and provided in the
+`errorMessage` attribute.
+
+The **error name** is not visible in the HTTP API response. The name for the
```
Comment (nikhil-varma) [inline site/content/3.12/develop/error-codes-and-meanings.md:40]: Same here

**Review by nikhil-varma — CHANGES_REQUESTED** [review]

**Comment by nikhil-varma** [comment]
@Simran-B CSS seems fine. Just some suggestions on the content. Please feel free to address it in the best suited approach. Thanks!

File: site/content/3.10/develop/error-codes-and-meanings.md [hunk 6]
```diff
@@ -2,9 +2,73 @@
...
+a unique name, an error message, and a description.
+
+The **error message** is a brief description of the error and provided in the
+`errorMessage` attribute.
+
+The **error name** is not visible in the HTTP API response. The name for the
+error with the number `11` is `ERROR_FORBIDDEN`, for instance. Error names are
+used in code to refer to an error kind, in the server's own code as well as in
+other code like drivers. In the [JavaScript API](javascript-api/_index.md), the
+`@arangodb` module maps error names to error numbers:
```
[the remark this answers is not in the document]
Comment (Simran-B) [reply site/content/3.10/develop/error-codes-and-meanings.md:44]: I did reorder this when writing twice but I think you are right that it's still not great. How about this order?

Error names are used in code to refer to an error kind, in the server's own code as well as in
other code like drivers. For instance, the name for the error with the number 11 is ERROR_FORBIDDEN.
The error name is not visible in the HTTP API response.

In the JavaScript API, ...

**Review by nerpaula — APPROVED** [review]

**Pull request merged**